src/Microsoft.AspNet.Mvc.Core/Filters/DefaultFilterProvider.cs

```diff
@@ -31,6 +34,10 @@ public void OnProvidersExecuting([NotNull] FilterProviderContext context)
                 {
                     ProvideFilter(context, item);
                 }
+                context.Results =
+                    context.Results
+                    .OrderBy(filter => filter.Descriptor, FilterDescriptorCorsComparer.Comparer)
```

Comment:
Wasn't involved in the design iterations, but it seems odd that filter providers need to be CORS-aware, no?